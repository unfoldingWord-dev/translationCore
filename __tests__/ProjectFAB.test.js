/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import ProjectFAB from '../src/js/components/home/projectsManagement/ProjectsFAB';

// Tests for ProjectFAB React Component
describe('Test ProjectFAB component', () => {
  test('Project FAB should match snapshot with options showing', () => {
    const renderedValue = renderer.create(
      <ProjectFAB
        translate={key => key}
        homeScreenReducer={{ showFABOptions: true }} />
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  test('Project FAB should match snapshot with options not showing', () => {
    const renderedValue = renderer.create(
      <ProjectFAB
        translate={key => key}
        homeScreenReducer={{ showFABOptions: false }} />
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
