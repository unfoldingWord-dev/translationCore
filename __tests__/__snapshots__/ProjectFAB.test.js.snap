// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProjectFAB component Comparing ProjectFAB Component render with snapshot taken 11/02/2017 in __snapshots__ should match 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-end",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "space-between",
      "zIndex": "999",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <table>
      <tbody>
        <tr>
          <td>
            <div
              style={
                Object {
                  "alignSelf": "flex-start",
                  "display": "flex",
                }
              }
            >
              <div
                onClick={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "alignItems": "center",
                    "alignSelf": "flex-start",
                    "backgroundColor": "#ffffff",
                    "borderRadius": "2px",
                    "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "cursor": "pointer",
                    "display": "flex",
                    "fontFamily": "Roboto, sans-serif",
                    "height": "40px",
                    "justifyContent": "center",
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "200px",
                    "zIndex": 1,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "padding": "0px",
                      "paddingBottom": 8,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.87)",
                        "fontSize": 14,
                        "muiPrepared": true,
                        "padding": "0px",
                      }
                    }
                  >
                    Import Local Project
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td>
            <div
              className={undefined}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "alignSelf": "flex-end",
                  "backgroundColor": "transparent",
                  "borderRadius": "50%",
                  "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23)",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "margin": "5px",
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "zIndex": "999",
                }
              }
            >
              <button
                disabled={false}
                href={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                onTouchTap={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                    "backgroundColor": "var(--accent-color-dark)",
                    "border": 10,
                    "borderRadius": "50%",
                    "boxSizing": "border-box",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "fontFamily": "Roboto, sans-serif",
                    "fontSize": "inherit",
                    "fontWeight": "inherit",
                    "height": 56,
                    "margin": 0,
                    "muiPrepared": true,
                    "outline": "none",
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "relative",
                    "textAlign": "center",
                    "textDecoration": "none",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "verticalAlign": "bottom",
                    "width": 56,
                    "zIndex": 1,
                  }
                }
                tabIndex={0}
                type="button"
              >
                <div
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    style={
                      Object {
                        "muiPrepared": true,
                        "top": 0,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      }
                    }
                  >
                    <span
                      className="glyphicon glyphicon-folder-open"
                      style={
                        Object {
                          "color": "#ffffff",
                          "fill": "#ffffff",
                          "fontSize": "26px",
                          "height": 56,
                          "lineHeight": "56px",
                        }
                      }
                    />
                  </div>
                </div>
              </button>
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <div
              style={
                Object {
                  "alignSelf": "flex-start",
                  "display": "flex",
                }
              }
            >
              <div
                onClick={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "alignItems": "center",
                    "alignSelf": "flex-start",
                    "backgroundColor": "#ffffff",
                    "borderRadius": "2px",
                    "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "cursor": "pointer",
                    "display": "flex",
                    "fontFamily": "Roboto, sans-serif",
                    "height": "40px",
                    "justifyContent": "center",
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "200px",
                    "zIndex": 1,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "padding": "0px",
                      "paddingBottom": 8,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.87)",
                        "fontSize": 14,
                        "muiPrepared": true,
                        "padding": "0px",
                      }
                    }
                  >
                    Import Online Project
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td>
            <div
              className={undefined}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "alignSelf": "flex-end",
                  "backgroundColor": "transparent",
                  "borderRadius": "50%",
                  "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23)",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "margin": "5px",
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "zIndex": "999",
                }
              }
            >
              <button
                disabled={false}
                href={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                onTouchTap={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                    "backgroundColor": "var(--accent-color-dark)",
                    "border": 10,
                    "borderRadius": "50%",
                    "boxSizing": "border-box",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "fontFamily": "Roboto, sans-serif",
                    "fontSize": "inherit",
                    "fontWeight": "inherit",
                    "height": 56,
                    "margin": 0,
                    "muiPrepared": true,
                    "outline": "none",
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "relative",
                    "textAlign": "center",
                    "textDecoration": "none",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "verticalAlign": "bottom",
                    "width": 56,
                    "zIndex": 1,
                  }
                }
                tabIndex={0}
                type="button"
              >
                <div
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    style={
                      Object {
                        "muiPrepared": true,
                        "top": 0,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      }
                    }
                  >
                    <span
                      className="glyphicon glyphicon-cloud-download"
                      style={
                        Object {
                          "color": "#ffffff",
                          "fill": "#ffffff",
                          "fontSize": "26px",
                          "height": 56,
                          "lineHeight": "56px",
                        }
                      }
                    />
                  </div>
                </div>
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <table>
    <tbody>
      <tr>
        <td>
          <div
            style={
              Object {
                "alignSelf": "flex-start",
                "display": "flex",
              }
            }
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#ffffff",
                  "borderRadius": "2px",
                  "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "pointer",
                  "display": "flex",
                  "fontFamily": "Roboto, sans-serif",
                  "height": "40px",
                  "justifyContent": "center",
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "200px",
                  "zIndex": 1,
                }
              }
            >
              <div
                style={
                  Object {
                    "padding": "0px",
                    "paddingBottom": 8,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontSize": 14,
                      "muiPrepared": true,
                      "padding": "0px",
                    }
                  }
                >
                  Close
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            className={undefined}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "alignSelf": "flex-end",
                "backgroundColor": "transparent",
                "borderRadius": "50%",
                "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23)",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "margin": "5px",
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "zIndex": "999",
              }
            }
          >
            <button
              disabled={false}
              href={undefined}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              onTouchTap={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "var(--reverse-color)",
                  "border": 10,
                  "borderRadius": "50%",
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 56,
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": "bottom",
                  "width": 56,
                  "zIndex": 1,
                }
              }
              tabIndex={0}
              type="button"
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  style={
                    Object {
                      "muiPrepared": true,
                      "top": 0,
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    }
                  }
                >
                  <span
                    className="glyphicon glyphicon-remove"
                    style={
                      Object {
                        "color": "var(--accent-color-dark)",
                        "fill": "#ffffff",
                        "fontSize": "26px",
                        "height": 56,
                        "lineHeight": "56px",
                      }
                    }
                  />
                </div>
              </div>
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <div>
    <div
      className=""
      style={
        Object {
          "border": "10px solid #ddd",
          "borderRadius": "50%",
          "boxSizing": "content-box",
          "height": 0,
          "left": -10,
          "position": "fixed",
          "top": -10,
          "transform": "translate(0px, 0px)",
          "transitionDuration": "1000ms",
          "transitionProperty": "all",
          "width": 0,
          "zIndex": 0,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "border": "1280px solid white",
            "borderRadius": "50%",
            "boxSizing": "content-box",
            "height": 20,
            "left": "50%",
            "opacity": 0.9,
            "position": "absolute",
            "top": "50%",
            "transform": "translate(-50%, -50%)",
            "transitionDuration": "1000ms",
            "transitionProperty": "all",
            "width": 20,
          }
        }
      />
    </div>
  </div>
</div>
`;
